xml/templates/dao.tpl

```diff
@@ -198,7 +198,8 @@ class {$table.className} extends CRM_Core_DAO {ldelim}
                       'permission'      => {$field.permission|@print_array},
 {/if}
 {if $field.default || $field.default === '0'}
-                         'default'   => '{if ($field.default[0]=="'" or $field.default[0]=='"')}{$field.default|substring:1:-1}{else}{$field.default}{/if}',
+  {capture assign=unquotedDefault}{if ($field.default[0]=="'" or $field.default[0]=='"')}{$field.default|substring:1:-1}{else}{$field.default}{/if}{/capture}
+                         'default'   => {if ($unquotedDefault==='NULL')}NULL{else}'{$unquotedDefault}'{/if},
```

Comment:
Looking at what you're doing here, and looking back through the history of this file, it seems pretty clear that this was simply an artifact of the XML having `NULL` in there and not needing to distinguish between that and a string because Pear::DB cleaned up after us.  I'm now enthusiastic about this, but I'm still nervous.